### PR TITLE
fix some portability issues (EHM-24)

### PR DESCRIPTION
**Pull request by Evlers**
## Testing

---

## Checklist

Before submitting a Pull Request, please ensure the following:

- [ ] 🚨 This PR does not introduce breaking changes.
- [ ] All CI checks (GH Actions) pass.
- [ ] Documentation is updated as needed.
- [ ] Tests are updated or added as necessary.
- [ ] Code is well-commented, especially in complex areas.
- [ ] Git history is clean — commits are squashed to the minimum necessary.

### CODE REVIEW

**Comment by mantriyogesh** [comment]
The changes look good. Anyway these changes are not specific to your port @Evlers , so should be fine to merge back to main.

Please give us some time to fetch and get these on main branch.

Again, thanks a lot for your efforts and time for submitting the PR.

**Comment by Evlers** [comment]
About _h_malloc_align&_h_free_align interfaces
For example, RT-Thread, aligned malloc must use aligned free for memory release
So it is necessary to add this interface

File: host/drivers/transport/transport_drv.c [hunk 1]
```diff
@@ -503,7 +507,7 @@ esp_err_t send_slave_config(uint8_t host_cap, uint8_t firmware_chip_id,
 	uint16_t len = 0;
 	uint8_t *sendbuf = NULL;
 
-	sendbuf = malloc(512); /*Arbitrary number*/
+	sendbuf = mempool_alloc(((struct mempool*)chan_arr[ESP_PRIV_IF]->memp), 512, true);
```
Comment (SohKamYung-Espressif) [inline host/drivers/transport/transport_drv.c:510]: This change does not look correct. It is allocating mempool memory from `char_arr`. `char_arr` is used for another purpose in ESP-Hosted.

File: host/drivers/mempool/mempool.c [hunk 2]
```diff
@@ -119,7 +119,7 @@ void * mempool_alloc(struct mempool* mp, int nbytes, int need_memset)
 #endif
 	}
 #else
-	buf = MEM_ALLOC(MEMPOOL_ALIGNED(nbytes));
+	buf = g_h.funcs->_h_malloc_align(MEMPOOL_ALIGNED(nbytes), MEMPOOL_ALIGNMENT_BYTES);
```
Comment (SohKamYung-Espressif) [inline host/drivers/mempool/mempool.c:122]: Have to add `_h_malloc_align` and `_h_free_align` to Hosted first.

File: host/drivers/transport/transport_drv.c (hunk 1 above)
Comment (Evlers) [reply host/drivers/transport/transport_drv.c:510]: Yes, it would be better to use g_h.funcs->_h_calloc

File: host/port/os_wrapper.c [hunk 3]
```diff
@@ -767,6 +787,8 @@ hosted_osi_funcs_t g_hosted_osi_funcs = {
 	._h_calloc                   =  hosted_calloc                  ,
 	._h_free                     =  hosted_free                    ,
 	._h_realloc                  =  hosted_realloc                 ,
+	._h_malloc_align			 =  hosted_malloc_align			   ,
+	._h_free_align				 =  hosted_free_align			   ,
```
Comment (SohKamYung-Espressif) [inline host/port/os_wrapper.c:791]: Not aligned correctly. Tabs before the `=` should be spaces.

File: host/hosted_os_adapter.h [hunk 4]
```diff
@@ -15,6 +15,8 @@ typedef struct {
 /* 4 */   void*  (*_h_calloc)(size_t blk_no, size_t size);
 /* 5 */   void   (*_h_free)(void* ptr);
 /* 6 */   void*  (*_h_realloc)(void *mem, size_t newsize);
+          void*  (*_h_malloc_align)(size_t size, rt_size_t align);
+          void   (*_h_free_align)(void* ptr);
```
Comment (SohKamYung-Espressif) [inline host/hosted_os_adapter.h:19]: `rt_size_t` -> `size_t`

For consistency, could add `/* 7 */` and `/* 8 */` here, although there is already a gap in the numbers.

**Comment by SohKamYung-Espressif** [comment]
@Evlers Looks good. Thanks for implementing the changes.

@mantriyogesh LGTM. Tested on P4 (SDIO interface), no issues found.

The PR Checklist says

> Git history is clean — commits are squashed to the minimum necessary.

So I think need to sqaush the commits into one commit.

**Comment by mantriyogesh** [comment]
That's okay.

We can take care @SohKamYung-Espressif of sequence.

**Pull request closed without merging**